Editorial: add CSS-AAM editor’s draft

ZoeBijl opened the pull request:
🚀 **Netlify Preview**:
🔄 **this PR updates the following sspecs**:
- [css-aam preview](https://deploy-preview-2673--wai-aria.netlify.app/css-aam/index.html) &mdash; [css-aam diff](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fw3c.github.io%2Fcss-aam%2F&doc2=https%3A%2F%2Fdeploy-preview-2673--wai-aria.netlify.app%2Fcss-aam%2Findex.html)

This description is a stub, please help expand it.

The creation of a CSS-AAM came up again during the tueseday ARIA session at TPAC ([minutes for this session](https://www.w3.org/2025/11/11-aria-minutes.html#23f4)). @jcsteh and I voulunteerd to be editors for this document.

There’s still work that needs to be done before this can be merged:

- [x] @spectranaut, @ZoeBijl, and (optionally) @jcsteh should chat with Rahim and Scott about taking over this work
- [x] @daniel-montalvo should discuss with strategy team, make sure were on the same page
- [x] @ZoeBijl should create an overview of existing work done around this (issues, discussions, etc)
    - Can be found on the [CSS-AAM wiki](https://github.com/w3c/css-aam/wiki/CSS%E2%80%90AAM-info-and-to-do-list#previous-work)
- [x] @ZoeBijl should clean up the old repository: [w3c/css-aam](https://github.com/w3c/css-aam) (pending https://github.com/w3c/css-aam/pull/12 and https://github.com/w3c/css-aam/pull/13)
    - [x] Purge issues
    - [x] Purge PR
    - [x] Purge branches
 - [x] Copy conformance section from core-aam

ZoeBijl committed:
feat: add core-css skeleton

ZoeBijl committed:
chore: update css-aam github workflow

based on Daniel’s feedback

ZoeBijl committed:
chore: implement feedback based on TPAC session

jcsteh commented on css-aam/README.md:
```diff
@@ -0,0 +1,5 @@
+
+CSS Accessibility API Mappings (css-aam) developed by the [Accessible Rich Internet Applications Working Group](https://www.w3.org/WAI/ARIA/) (previously the [Web Applications Working Group](https://www.w3.org/2019/webapps/)).
+
+This specification was part of the [ARIA suite](https://www.w3.org/WAI/ARIA/deliverables). General information about editing the specification is in the [main ARIA repository readme](https://github.com/w3c/aria/).
```
nit: Should this be "is" instead of "was"?

jcsteh commented on css-aam/index.html:
```diff
@@ -0,0 +1,169 @@
+                // Editors
+                editors: [
+                    { name: "Zoë Bijl", company: "Invited Expert", mailto: "editor@moiety.me", w3cid: 74040 },
+                    { name: "James Teh", company: "Mozilla", companyURL: "https://mozilla.org/" },
```
Strictly speaking, I work for Mozilla Corporation, but our W3C membership is under Mozilla Foundation, so...

```suggestion
                    { name: "James Teh", company: "Mozilla Foundation", companyURL: "https://mozilla.org/", mailto: "jteh@mozilla.com", w3cid: 134488 },
```

jcsteh commented on css-aam/index.html:
```diff
@@ -0,0 +1,169 @@
+                <h3>General Rules for Exposing WAI-ARIA Semantics</h3>
+            </section>
+            <section id="css-property-mappings">
+                <h3>CSS Property mappings</h3>
```
nit: Title case.
```suggestion
                <h3>CSS Property Mappings</h3>
```

jcsteh commented on css-aam/index.html:
```diff
@@ -0,0 +1,169 @@
+                <h3>CSS Property mappings</h3>
+            </section>
+            <section id="css-pseudo-element-mappings">
+                <h3>CSS Pseudo-element mappings</h3>
```
nit: Title case.
```suggestion
                <h3>CSS Pseudo-element Mappings</h3>
```

jcsteh commented on css-aam/index.html:
```diff
@@ -0,0 +1,169 @@
+        </section>
+        <section>
+            <h2>Mapping CSS to Accessibility APIs</h2>
+            <section id="css-mapping">
```
```suggestion
            <section id="mapping_general">
```

jcsteh reviewed (changes requested):
Thanks so much for getting this started. Sorry it took me a while to get to this.

ZoeBijl committed:
correct Jamie’s details

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl commented on css-aam/.editorconfig:
```diff
@@ -0,0 +1,9 @@
+charset = utf-8
+end_of_line = lf
+indent_style = space
+indent_size = 4
```
i’d like to discuss this tonight during the editor’s meeting. i prefer 4 spaces for readability but not a hill i’d like to die on. @jcsteh do you have a preference?

ZoeBijl replied:
RESOLUTION: CSS-AAM will stick with 4 spaces for indenting. Willing to change if the ARIA WG decides consistency is more important.

daniel-montalvo committed:
Try skipping prettier at this stage

daniel-montalvo committed:
Revert "Try skipping prettier at this stage" If we change this we'll do on another PR.

This reverts commit 62913d492c11da08592ea859ff6c836cc3df7fd5.

pkra committed:
Merge branch 'main' into css-aam

pkra commented:
I updated the branch to fix the false positive failure in the roleInfo check.

daniel-montalvo committed:
Merge branch 'main' into css-aam

daniel-montalvo committed:
Merge branch 'main' into css-aam

daniel-montalvo committed:
Merge branch 'main' into css-aam

ZoeBijl committed:
fix: remove second paragraph

this was decided in https://github.com/w3c/aria/pull/2673#discussion_r2560192494

ZoeBijl committed:
fix: update copyright date

ZoeBijl committed:
fix: remove collaboration paragraph

nothing from the old doc has been used for this iteration

ZoeBijl committed:
fix: title case

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl committed:
fix: title case

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl committed:
fix: section ID

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl committed:
feat: add section on default mappings

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl committed:
fix: section id

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl commented on css-aam/index.html:
```diff
@@ -0,0 +1,169 @@
+                <h3>CSS Pseudo-element mappings</h3>
+            </section>
+            <section id="css-pseudo-element-mappings">
+                <h3>CSS Mapping pseudo-class selectors</h3>
```
- [ ] add clarification/example for line 167.

ZoeBijl committed:
feat: add css pseudo-class mapping section

Co-authored-by: James Teh <jamie@jantrid.net>

ZoeBijl committed:
Merge branch 'main' into css-aam

ZoeBijl commented:
Just to check: is there anything @jcsteh and I need to do to get this merged?

spectranaut reviewed (approved):
Sorry this took me so long to take a look -- but this looks great. I spoke with @daniel-montalvo and I think we should merge it.

In response to @pkra's question I'd say this is editorial.
